### PR TITLE
Include 'if not exists' when creating index

### PR DESCRIPTION
**Pull request by octonato**
References #488 and #489 

We need extra tests, like running the utility twice to make sure that everything is good for all DBs.

### CODE REVIEW

**Comment by hseeberger** [comment]
That is not enough: the subsequent PK creation is also erroneous. See my other issue you closed, @octonato .

File: akka-projection-jdbc/src/main/scala/akka/projection/jdbc/internal/Dialect.scala [hunk 1]
```diff
@@ -51,7 +51,7 @@ private[projection] object DialectDefaults {
       "LAST_UPDATED" BIGINT NOT NULL
```
Comment (hseeberger) [inline akka-projection-jdbc/src/main/scala/akka/projection/jdbc/internal/Dialect.scala:51]: You also have to delete the creation of the PK below and include it here in the conditional table creation:

```
    primary key ("PROJECTION_NAME", "PROJECTION_KEY")
```

**Review by hseeberger — CHANGES_REQUESTED** [review]

File: akka-projection-jdbc/src/main/scala/akka/projection/jdbc/internal/Dialect.scala (hunk 1 above)
Comment (octonato) [reply akka-projection-jdbc/src/main/scala/akka/projection/jdbc/internal/Dialect.scala:51]: I will incorporate this. Thanks

Comment (octonato) [reply akka-projection-jdbc/src/main/scala/akka/projection/jdbc/internal/Dialect.scala:51]: I don't remember why the PK is added on another statement. Anyway, if there is an issue we will find out when we change it.

Comment (hseeberger) [reply akka-projection-jdbc/src/main/scala/akka/projection/jdbc/internal/Dialect.scala:51]: It certainly works for Postgress (current version) with the PK creation in the table creation.

**Review by seglo — COMMENTED** [review]
Should we do the same for the other dialects?

**Comment by octonato** [comment]
> Should we do the same for the other dialects?

Yes, we should. We need to test them all. I don't remember which one has it already Oracle of SqlServer.

File: akka-projection-slick/src/main/scala/akka/projection/slick/internal/SlickOffsetStore.scala [hunk 2]
```diff
@@ -113,7 +113,7 @@ import slick.jdbc.JdbcProfile
 
   val offsetTable = TableQuery[OffsetStoreTable]
 
-  def createIfNotExists: Future[Done] = {
+  def createIfNotExists(): Future[Done] = {
```
Comment (octonato) [inline akka-projection-slick/src/main/scala/akka/projection/slick/internal/SlickOffsetStore.scala:116]: side-effect method, using `()`. 

This is internal API, so no big deal to change it.

**Comment by octonato** [comment]
hmm, that PR is not building. Probably because it's not targeting `master`, but another PR branch. When #491 gets merged, it will change to `master` and will probably build. 

That also explain why this is always turning green and #491 is failing.

**Review by hseeberger — APPROVED** [review]
LGTM, has all the changes I was looking for. Also looks comprehensive and consistent.

I am not sure if I like the `Await` style tests, but I guess these are there for historical reasons.

**Comment by octonato** [comment]
> I am not sure if I like the Await style tests, but I guess these are there for historical reasons.

Same feeling here, but indeed there is a reason for that. 

We could have used `.futureValue` (which is a disguised Await) and the `PatienceConfig`, but we have experience much high timeouts when creating DBs then when doing some other general async tests. We then decided to have a explicit and much larger timeout number for the db creation.

That said, I think we could use `.futureValue` in that case. This is just a DB call like any other call that's is done during test. The create db call we do in the setup should use a larger timeout value because it's really creating the schema.

File: akka-projection-jdbc/src/main/scala/akka/projection/jdbc/internal/Dialect.scala [hunk 3]
```diff
@@ -132,21 +140,25 @@ private[projection] case class MySQLDialect(schema: Option[String], tableName: S
...
+        CURRENT_OFFSET VARCHAR(255) NOT NULL,
+        MANIFEST VARCHAR(4) NOT NULL,
+        MERGEABLE BOOLEAN NOT NULL,
+        LAST_UPDATED BIGINT NOT NULL,
+        PRIMARY KEY(PROJECTION_NAME,PROJECTION_KEY)
+       );
     """,
       // create index
-      s"""CREATE INDEX IF NOT EXISTS PROJECTION_NAME_INDEX ON $table (PROJECTION_NAME);""")
+      s"""CREATE INDEX PROJECTION_NAME_INDEX ON $table (PROJECTION_NAME);""")
```
Comment (octonato) [inline akka-projection-jdbc/src/main/scala/akka/projection/jdbc/internal/Dialect.scala]: We are not yet fully telling the truth when we call the MySQL create table script from one of the methods named `createIfNotExists`. The index creation will fail. 

There are means to cover that in MySQL, but not so straightforward (see https://dba.stackexchange.com/questions/24531/mysql-create-index-if-not-exists). 

For the moment the way out is to call `dropIfExists` and then `createIfNotExists`. Which is pretty much ok for tests purposes.

File: akka-projection-jdbc/src/main/scala/akka/projection/jdbc/internal/Dialect.scala [hunk 4]
```diff
@@ -48,14 +50,14 @@ private[projection] object DialectDefaults {
...
       // create index
-      s"""CREATE INDEX "PROJECTION_NAME_INDEX" on $table ("PROJECTION_NAME");""",
-      // add primary key
-      s"""ALTER TABLE $table
-       ADD CONSTRAINT "PK_PROJECTION_ID" PRIMARY KEY("PROJECTION_NAME","PROJECTION_KEY");
-    """)
+      s"""CREATE INDEX IF NOT EXISTS "PROJECTION_NAME_INDEX" on $table ("PROJECTION_NAME");""")
+
+  def dropTableStatement(table: String): String =
+    s" DROP TABLE IF EXISTS $table;"
```
Comment (octonato) [inline akka-projection-jdbc/src/main/scala/akka/projection/jdbc/internal/Dialect.scala:60]: we must be able to drop the offset in tests, see https://github.com/akka/akka-platform-guide/pull/422#issuecomment-745551261

**Review by ennru — APPROVED** [review]
LGTM.

**Pull request merged**